packages/core/src/browser/navigatable.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (C) 2018 Red Hat, Inc. and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import URI from '../common/uri';
+
+/**
+ * Each widget which holds an uri to a workspace file and wants to be able to reveal that file in navigator,
```

Comment:
Let's leave it as is.

But I see other use cases for `Navigatable`, for example, to open an editor for any navigatable widget or selection. In that case, a target URI should not be necessary a file URI, a navigatable is not a necessary widget, a client is not a widget as well.

I concern that it is too narrow and introduces a backward dependency from the core to the navigator extension if i read a comment as a contract.

Reply:
Yes it purpose indeed much wider then I described. Thanks for pointing out.